src/compiler/program.ts

```diff
@@ -1168,9 +1168,6 @@ namespace ts {
                             let typeAssertionExpression = <TypeAssertion>node;
                             diagnostics.push(createDiagnosticForNode(typeAssertionExpression.type, Diagnostics.type_assertion_expressions_can_only_be_used_in_a_ts_file));
                             return true;
-                        case SyntaxKind.Decorator:

```

Comment:
My only strong feeling is that showing an error is too much, showing a warning that this is an experimental feature would be more appropriate. But then the warning has to tell me how to get rid of it.